prow/plank/reconciler.go

```diff
@@ -344,12 +344,23 @@ func (r *reconciler) defaultReconcile(ctx context.Context, request reconcile.Req
 		// Objects can be deleted from the API while being in our workqueue
 		return reconcile.Result{}, nil
 	}
+	originalPJ := pj.DeepCopy()
 
 	res, err := r.serializeIfNeeded(ctx, pj)
 	if IsTerminalError(err) {
 		// Unfixable cases like missing build clusters, do not return an error to prevent requeuing
-		r.log.WithError(err).WithFields(pjutil.ProwJobFields(pj)).
-			Error("Reconciliation failed with terminal error and will not be requeued")
+		log := r.log.WithError(err).WithFields(pjutil.ProwJobFields(pj))
+		log.Error("Reconciliation failed with terminal error and will not be requeued")
+		if !pj.Complete() {
+			pj.SetComplete()
+			pj.Status.State = prowv1.ErrorState
+			pj.Status.Description = fmt.Sprintf("Terminal error: %v.", err)
```

Comment:
Should we always set these fields like this, even if pj.Complete() is true? I would think at least pj.Status.Description would be always set to "Terminal error: ...".

Reply:
The pattern throughout plank (which ideally would be enforced/codified as a function) is that the state and description are set whenever the job is marked complete.  So if the job is already complete these fields will be set.

Reply:
I guess what I'm asking is, if we encounter a TerminalError() in this codepath, should we always mark the pj.Status.Description as such, even if the job was already (somehow) previously marked as complete.

OTOH since you've mentioned that the description marches in lockstep with the job completion, perhaps my question is moot.

Reply:
Makes sense. Thanks